### PR TITLE
Template rework [op-145]

### PR DESCRIPTION
**Pull request by Rytiggy**
This Commit:

- restructures the data structures to be based off one unified settings blob. 
- Rebuilds all templates so that they are based off the new settings 
- abstracts Layouts themes and user settings.
- QA on all code 

**Creating a new theme**
_public/defaultThemes/_
create a new theme in this file in the below structure 
``` 
{
	"id": 2,
	"name": "Orange Tree",
	"theme" : {
		"primaryColor" : "#f07057",
		"primaryTextColor" : "snow",
		"secondaryColor" : "snow",
		"secondaryTextColor" : "#black",
		"tertiaryColor" : "#eeeeee",
		"tertiaryTextColor" : "black",
		"alignment": "center",
		"blockSettings" : {
			"layer-title" :{
				"backgroundImage":"https://image.ibb.co/cqBWUw/bg.png"
			}
	}
}

```

**renderer defaultTheme.js** 
_add name of theme to themeList at top of page_ 
` let  themeList = [ 'theme_1.json', 'theme_2.json' , 'theme_3.json' , 'theme_4.json']`

**Create a new template** 
_Layouts.js_ 
Add new layout 
`"landing-page-v0.0.1": [ "layer-title-v0.0.1", "layer-info-v0.0.1" , "layer-file-v0.0.1"]
`
**Layer-template-select.js**
_Add to the templatesList (this makes it show up on the page )_
```
{
    'layoutName': 'Data Feature', //Name on page
    'layout': 'data-feature-v0.0.1',  //Layout with version for use in code 
    'themeId' : 2 , // Theme ID  (theme id 2 would get theme_3.json [0-index])
    'url' : ENV.rootURL +'img/green.png', //Link to image used to show what theme looks like
    'colors':themes[2].theme // pulls in colors from theme to show on page to user
}
```

### CODE REVIEW

File: app/components/add-layer/component.js [hunk 1]
```diff
@@ -7,152 +11,62 @@ export default Ember.Component.extend({
...
-                        "component": "layer-settings",
-                        "wikiId": "",
-                        "sectionTitle": "Wiki example",
-                        "showInNavigation": true,
-                        "addShowMore": false,
-                        "bgColor": "#FFFFFF",
-                        "color": "#333333",
-                        "alignment": "center"
-                    }
+                    "settings": tempSettings['layer-wiki']
```
Comment (birdbrained) [inline app/components/add-layer/component.js]: Can we make this this an object?

File: app/components/layer-file/template.hbs [hunk 2]
```diff
@@ -33,10 +33,11 @@
 
     {{#if layer.settings.showDownload}}
         {{#if editMode}}
-         <a class="btn pull-right btn-info btn-outline-cus btn-lg" style="color:#5bc0de">{{input value=layer.settings.buttonText }}
+         <a class="btn pull-right btn-info btn-outline-cus btn-lg" style="color:#5bc0de; margin-bottom: 20px;">{{input value=layer.settings.buttonText }}
   {{fa-icon "download"}}</a>
         {{else}}
-            <a class="btn pull-right btn-info btn-outline-cus btn-lg" href={{layer.settings.downloadLink}} >{{layer.settings.buttonText }}{{fa-icon "download"}}</a>
+            <a class="btn pull-right btn-info btn-outline-cus btn-lg" href={{layer.settings.downloadLink}} style="margin-bottom: 20px;">{{layer.settings.buttonText }}{{fa-icon "download"}}</a>
```
Comment (birdbrained) [inline app/components/layer-file/template.hbs]: Move this out to a stylesheet?

File: app/components/layer-settings/template.hbs [hunk 3]
```diff
@@ -126,7 +126,7 @@
                 {{{get helpText layer.component}}}
                 {{#if OSFcontent}}
                     <br>
-                    <a class='' href="http://www.osf.io/{{node.id}}" taget="_blank" >View/Change Content</a>
+                    <a class='' href="http://www.osf.io/{{node.id}}" target="_blank" >View/Change Content</a>
```
Comment (birdbrained) [inline app/components/layer-settings/template.hbs]: Niiiice

File: app/components/layer-template-select/component.js [hunk 4]
```diff
@@ -3,78 +3,201 @@
...
+    'colors':themes[2].theme
+}
+];
+
+function compareUserSettings(layerType ,systemSettings , userSettings) { 
+    const ignoredSettings = ['backgroundImage' , 'bgColor' , 'color' , 'alignment']
+    let newSettings = {};
+
+
+    for(let i = 0; i < userSettings.length; i++) { //Loop over users settings
```
Comment (birdbrained) [inline app/components/layer-template-select/component.js]: conde style; needs spacing

File: app/components/layer-template-select/component.js [hunk 5]
```diff
@@ -3,78 +3,201 @@
...
+                    Ember.set( newSetting, "color",  theme.tertiaryTextColor); 
+                    Ember.set( newSetting, "alignment",  theme.alignment); 
+                }
+
+
+                //Block specific settings
+                if(layerType === 'layer-title') {
+                   Ember.set( newSetting, "bgColor",  theme.primaryColor); 
+                   Ember.set( newSetting, "color",  theme.primaryTextColor); 
+                   Ember.set( newSetting, "alignment",  theme.alignment);
```
Comment (birdbrained) [inline app/components/layer-template-select/component.js]: Some more code style/spacing issues

File: app/components/layer-template-select/style.scss [hunk 6]
```diff
@@ -32,11 +32,47 @@
...
+  position:relative;
+  overflow:hidden;
+  background:#666;
+ 
+
+  
+  -webkit-transform: rotate(45deg);
+  -moz-transform: rotate(45deg);
+  -ms-transform: rotate(45deg);
+    -o-transform: rotate(45deg);
```
Comment (birdbrained) [inline app/components/layer-template-select/style.scss]: A bit more spacing

File: app/components/layer-theme-select/component.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+	isOpen: null,
+	classNames: ['d-inline'],
+	themeList: [ 
+		{
+			id: 1,
+			name: 'Blue sky',
+			colorOne: '#31708f',
+			colorTwo: 'snow',
+			colorThree: '#98cbe4' 
+		},
```
Comment (birdbrained) [inline app/components/layer-theme-select/component.js]: Check on the spacing

File: app/components/layer-theme-select/template.hbs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+		<div class="container">
+			<div class="overlay-dismiss">
+				<i class="material-icons" {{action 'dismiss'}}>close</i>
+			</div>
+			<h2  class="btn-margin-bottom"> Select a theme</h2>
+			{{layer-template-select model=model theme=model.theme showButton=true }}
+			<div class="row p-v-xl">
+				{{#each themeList as |item|}}
+				<div class="col-sm-3 m-b-lg">
+					<div class="theme-select-item" {{action 'selectTheme' item.id}}>
```
Comment (birdbrained) [inline app/components/layer-theme-select/template.hbs]: indentation

File: app/renderer/defaultThemes.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+themeList.forEach((themeName)=> {
+  getThemes(themeName)
+});
+
+
+
+
+export {
+    themes
```
Comment (birdbrained) [inline app/renderer/defaultThemes.js]: spacing and newline at the end

File: app/components/layer-template-select/component.js [hunk 10]
```diff
@@ -3,78 +3,201 @@
...
+
+//import * as util from "./util";
+
+
+
+function isEven(value) {
+    return (value%2 == 0);
+}
+//layoutKey , themeId)
+let templatesList = [
```
Comment (cameronblandford) [inline app/components/layer-template-select/component.js]: cleaning up indentation here would be a nice touch

File: app/components/layer-template-select/component.js [hunk 11]
```diff
@@ -3,78 +3,201 @@
...
+    'themeId' : 2 , 
+    'url' : ENV.rootURL +'img/green.png',
+    'colors':themes[2].theme
+}
+];
+
+function compareUserSettings(layerType ,systemSettings , userSettings) { 
+    const ignoredSettings = ['backgroundImage' , 'bgColor' , 'color' , 'alignment']
+    let newSettings = {};
+
```
Comment (cameronblandford) [inline app/components/layer-template-select/component.js]: remove a line here

File: app/renderer/layouts.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+
+let layouts = {
+	"landing-page-v0.0.1": [ "layer-title-v0.0.1", "layer-info-v0.0.1" , "layer-file-v0.0.1"],
+	"research-paper-v0.0.1": [ "layer-title-v0.0.1", "layer-wiki-v0.0.1", "layer-info-v0.0.1" , "layer-file-v0.0.1"],
+	"data-feature-v0.0.1": [ "layer-title-v0.0.1" , "layer-advanced-v0.0.1", "layer-file-v0.0.1", "layer-file-v0.0.1"]
+
+}
+
```
Comment (cameronblandford) [inline app/renderer/layouts.js]: too many newlines

**Pull request merged**